### DISCUSSION TITLE
Is there a way to check the status of rabbitmq node data synchronization ?

### DISCUSSION THREAD

**Discussion by xixiangzouyibian** [rabbitmq/cluster-operator#1502] [Q&A] [answered]
For example: 
We deployed 3 rabbitmq nodes, each rabbitmq node deployed on one k8s node (anti-affinity). 
Now one of rabbitmq node broken for some reasons, we migrate the broken rabbitmq node to idle k8s node then we get a healthy three-node rabbitmq cluster. 
But from my understanding, the new deployed rabbitmq node will take the time to do the fully data synchronization from master rabbitmq node.  so how do I know if data sync is finish ? 
It seems query rabbitmqcluster may not work:
> NAME       ALLREPLICASREADY   RECONCILESUCCESS   AGE
> rabbitmq   True                            True                             4d

But elasticsearch provides the field(HEALTH) to query:

> NAME      HEALTH   NODES   VERSION   PHASE   AGE
> catalog     green      3             7.17.10      Ready    4d
> search      green      3             7.17.10      Ready    4d

**Comment by xixiangzouyibian** [comment]
Any updates ?

**Reply by lukebakken** [reply]
It is impolite to bump your own question. It has only been four days, be patient.

**Comment by Zerpet** [comment]
There's no practical way to verify this. Most of the time you don't need to worry about this. RabbitMQ will do it's replication and it will complete relatively fast. If you want to inspect a specific queue, you have the command [rabbitmq-queues quorum_status](https://rabbitmq.com/rabbitmq-queues.8.html#quorum_status) or [rabbitmq-streams stream_status](https://rabbitmq.com/rabbitmq-streams.8.html#stream_status)

**Reply by xixiangzouyibian** [reply]
@Zerpet Thanks for you response.
Is there any data that can indicate that the data synchronization between nodes is fast ? Could you evaluate the magnitude of the synchronization time, seconds level or millisec level ? And will the synchronization time become longer when there is more data, more queues or more rabbitmq nodes.

**Reply by michaelklishin** [reply]
[Infrastructure-level metrics](https://www.rabbitmq.com/monitoring.html#system-metrics) can.

**Comment by michaelklishin** [comment, marked as answer]
[Starting with RabbitMQ 3.13](https://github.com/rabbitmq/rabbitmq-server/pull/9162), new quorum queue replicas will not be considered "joined" (online, counting towards the quorum) if they are performing their initial sync. All subsequent syncs should transfer drastically less data (well, proportionally to how far behind this replica is, of course).

**Reply by xixiangzouyibian** [reply]
Thanks Michael.
